draft-ietf-gnap-core-protocol.md

```diff
@@ -6223,6 +6301,39 @@ instance capable of multiple formats can use [AS discovery](#discovery) to deter
 are supported, if desired. An AS should be generous in supporting many different key formats to
 allow different types of client software and client instance deployments.
 
+## Asynchronous Interactions {#security-async}
+
+GNAP allows the RO to be contacted by the AS asynchronously, outside the regular flow of the
+protocol. This allows for some advanced use cases, such as cross-user authentication or information
+release, but such advanced use cases have some distinct issues that implementors need to be fully
+aware of before using these features.
+
+First, in many applications, the return of a subject information to the client instance could
+indicate to the client instance that the end-user is the party represented by that information,
+functionally allowing the end-user to authenticate to the client application. While the details of
+a fully functional authentication protocol are outside the scope of GNAP, it is a common
+exercise for a client instance to be requesting information about the end user. This is facilitated
+by the several [interaction methods](#interaction-start) defined in GNAP that allow the end user
+to begin interaction directly with the AS. However, when the subject of the information is
+intentionally not the end-user, the client application will need some way to differentiate between
+requests for authentication of the end user and requests for information about a different user.
+Confusing these states could lead to an attacker having their account associated with a privileged
+user. Client instances can mitigate this by having distinct code paths for primary end user
+authentication and requesting subject information about secondary users, such as in a call center.
```

Comment:
a typical use case is the support of a client application (e.g. mobile), while the agent uses a corporate PC (e.g. desktop app). So in some cases, the client class_id might help too